usr.bin/sockstat/sockstat.c

```diff
@@ -1107,235 +1093,420 @@ sctp_path_state(int state)
+				s->proto == IPPROTO_UDP) {
+				len = snprintf(NULL, 0,
+					"%" PRIu64, s->inp_gencnt);
+				if (len > cw->inp_gencnt)
+					cw->inp_gencnt = len;
 			}
-			offset += 9;
 		}
 		if (opt_U) {
 			if (faddr != NULL &&
-			    ((s->proto == IPPROTO_SCTP &&
-			      s->state != SCTP_CLOSED &&
-			      s->state != SCTP_BOUND &&
-			      s->state != SCTP_LISTEN) ||
-			     (s->proto == IPPROTO_TCP &&
-			      s->state != TCPS_CLOSED &&
-			      s->state != TCPS_LISTEN))) {
-				do
-					pos += xprintf(" ");
-				while (pos < offset);
-				pos += xprintf("%u",
-				    ntohs(faddr->encaps_port));
+				((s->proto == IPPROTO_SCTP &&
+					s->state != SCTP_CLOSED &&
+					s->state != SCTP_BOUND &&
+					s->state != SCTP_LISTEN) ||
+					(s->proto == IPPROTO_TCP &&
+					s->state != TCPS_CLOSED &&
+					s->state != TCPS_LISTEN))) {
+				len = snprintf(NULL, 0, "%u",
+					ntohs(faddr->encaps_port));
+				if (len > cw->encaps)
+					cw->encaps = len;
 			}
-			offset += 7;
 		}
 		if (opt_s) {
 			if (faddr != NULL &&
-			    s->proto == IPPROTO_SCTP &&
-			    s->state != SCTP_CLOSED &&
-			    s->state != SCTP_BOUND &&
-			    s->state != SCTP_LISTEN) {
-				do
-					pos += xprintf(" ");
-				while (pos < offset);
-				pos += xprintf("%s",
-				    sctp_path_state(faddr->state));
+				s->proto == IPPROTO_SCTP &&
+				s->state != SCTP_CLOSED &&
+				s->state != SCTP_BOUND &&
+				s->state != SCTP_LISTEN) {
+				len = snprintf(NULL, 0, "%s",
+					sctp_path_state(faddr->state));
+				if (len > cw->path_state)
+					cw->path_state = len;
 			}
-			offset += 13;
 		}
 		if (first) {
 			if (opt_s) {
 				if (s->proto == IPPROTO_SCTP ||
-				    s->proto == IPPROTO_TCP) {
-					do
-						pos += xprintf(" ");
-					while (pos < offset);
+					s->proto == IPPROTO_TCP) {
 					switch (s->proto) {
 					case IPPROTO_SCTP:
-						pos += xprintf("%s",
-						    sctp_conn_state(s->state));
+						len = snprintf(NULL, 0, "%s",
+							sctp_conn_state(s->state));
+						if (len > cw->conn_state)
+							cw->conn_state = len;
 						break;
 					case IPPROTO_TCP:
 						if (s->state >= 0 &&
-						    s->state < TCP_NSTATES)
-							pos += xprintf("%s",
-							    tcpstates[s->state]);
+							s->state < TCP_NSTATES) {
+							len = snprintf(NULL, 0, "%s",
+								tcpstates[s->state]);
+							if (len > cw->conn_state)
+								cw->conn_state = len;
+						}
+						break;
+					}
+				}
+			}
+		}
+		if (laddr != NULL)
+			laddr = laddr->next;
+		if (faddr != NULL)
+			faddr = faddr->next;
+		first = false;
+	}
+}
+
+static void
+calculate_column_widths(struct col_widths *cw)
+{
+	cw->user		= 4;
+	cw->command		= 10;
+	cw->pid			= 3;
+	cw->fd			= 2;
+	cw->proto		= 5;
+	cw->local_addr	= 13;
+	cw->foreign_addr	= 15;
+	cw->pcb_kva		= 18;
+	cw->fib			= 3;
+	cw->splice_address	= 14;
+	cw->inp_gencnt	= 2;
+	cw->encaps		= 6;
+	cw->path_state	= 10;
+	cw->conn_state	= 10;
+	cw->stack		= TCP_FUNCTION_NAME_LEN_MAX;
+	cw->cc			= TCP_CA_NAME_MAX;
```

Comment:
I suggest not trying to align the parts after the `=`.  But if you do, make sure you use spaces, or use tabstop=8.  As it is, they aren't actually aligned.